lib/any.js

```diff
@@ -683,6 +683,21 @@ module.exports = internals.Any = class {
             }
         }
 
+        if (this._settings) {
+
+            const optionSet = ['abortEarly', 'convert', 'allowUnknown', 'skipFunctions', 'stripUnknown', 'language', 'presence', 'context', 'noDefaults'];
+            if (optionSet.some((option) => this._settings.hasOwnProperty(option))) {
```

Comment:
@Marsup sorry, crazy busy at work the last week.

`.when` is using `this._settings` to save a reference to the baseType [here](https://github.com/hapijs/joi/blob/master/lib/any.js#L381). That's the main issue.

The settings I want to preserve are set [by .options here](https://github.com/hapijs/joi/blob/master/lib/any.js#L207) and [by .strict here](https://github.com/hapijs/joi/blob/master/lib/any.js#L215).

Now that I look at it more, I think I can remove the filtering and just check to make sure that `baseType` is not the only value and that it doesn't get set. I'll take a stab at that as soon as I get a chance.